### PR TITLE
refactor resnet50

### PR DESCRIPTION
**Pull request by shh2000**
log is running
expectation:
4.4k ips(using bs==32,lr=0.1, acc1>75.5, the same as torchvision)
(if turn bs -> 256, using 28/40 GiB memory, need search a hyperparameter lr. ifnot, acc = 73.3, not 75.5. bs=256 get 5.8k ips(raw) and 9.2k ips(without IO time))
searching suitable lr for bs==256 can handle later

### CODE REVIEW

File: training/benchmarks/resnet50/pytorch/train/trainer.py [hunk 1]
```diff
@@ -1,239 +1,144 @@
...
+import torch.distributed as dist
+
+from model import create_model
+from optimizers import create_optimizer
+from schedulers import create_scheduler
+from train.evaluator import Evaluator
+from train.training_state import TrainingState
+
+CURR_PATH = os.path.abspath(os.path.dirname(__file__))
+sys.path.append(os.path.abspath(os.path.join(CURR_PATH, "../../")))
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer.py:21]: 这两句可以删除

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 2]
```diff
@@ -1,176 +1,134 @@
...
             "e2e_time": e2e_time,
-            "global_steps": state.global_steps,
-            "training_images_per_second": training_perf,
+            "train_time": state.traintime,
+            "train_no_eval_time": state.noevaltime,
+            "pure_training_computing_time": state.purecomputetime,
+            "throuhgput(ips)_raw": state.num_trained_samples / state.traintime,
+            "throuhgput(ips)_no_eval":
+            state.num_trained_samples / state.noevaltime,
+            "throuhgput(ips)_pure_compute":
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:128]: typo:  throughput

File: training/benchmarks/resnet50/pytorch/train/training_state.py [hunk 3]
```diff
@@ -1,84 +1,40 @@
...
+    acc1: float = 0.0
+
+    epoch: int = 1
+
+    end_training: bool = False
+    converged: bool = False
+
+    traintime = 0.0
+    noevaltime = 0.0
+    purecomputetime = 0.0
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/training_state.py:27]: 建议用snake_case来提升可读性。涉及这几个变量：traintime，noevaltime，purecomputetime

File: training/benchmarks/resnet50/pytorch/train/trainer.py [hunk 4]
```diff
@@ -1,239 +1,144 @@
...
+            else:
+                output = model(images)
+
+                criterion = torch.nn.CrossEntropyLoss()
+                loss = criterion(output, target)
+                loss.backward()
+                optimizer.step()
+
+            if step % self.config.log_freq == 0:
+                print("Train Step " + str(step) + "/" + str(len(data_loader)) +
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer.py:102]: 用f-string就不用进行类型转换了

File: training/benchmarks/resnet50/pytorch/train/trainer.py [hunk 5]
```diff
@@ -1,239 +1,144 @@
...
+            state.end_training = True
+        state.num_trained_samples += len(data_loader.dataset)
+
+    @torch.no_grad()
+    def evaluate(self, model, data_loader, device):
+        acc1_total = 0.0
+        steps = 0
+        for step, batch in enumerate(data_loader):
+            if step % self.config.log_freq == 0:
+                print("Eval Step " + str(step) + "/" + str(len(data_loader)))
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer.py:133]: 同上

File: training/benchmarks/resnet50/pytorch/train/trainer_adapter.py [hunk 6]
```diff
@@ -1,36 +1,43 @@
...
+import torch
+
+import torch.distributed as dist
+from torch.optim import Optimizer
+from torch import nn, Tensor
+from torch.nn.parallel import DistributedDataParallel as DDP
+import config
+
+CURR_PATH = os.path.abspath(os.path.dirname(__file__))
+sys.path.append(os.path.abspath(os.path.join(CURR_PATH, "../../../")))
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer_adapter.py:15]: 这两句可以去掉

File: training/benchmarks/resnet50/pytorch/train/trainer_adapter.py [hunk 7]
```diff
@@ -1,36 +1,43 @@
...
+from driver.dist_pytorch import main_proc_print
+
+
+def convert_model(model: nn.Module) -> nn.Module:
+    """convert_model"""
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer_adapter.py:20]: rm foo comments

File: training/benchmarks/resnet50/pytorch/train/trainer_adapter.py [hunk 8]
```diff
@@ -1,36 +1,43 @@
...
+    return model
+
+
+def model_to_fp16(model: nn.Module) -> nn.Module:
+    """model_to_fp16"""
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer_adapter.py:25]: 同上

File: training/benchmarks/resnet50/pytorch/train/trainer_adapter.py [hunk 9]
```diff
@@ -1,36 +1,43 @@
...
+    # To prevent OOM for model sizes that cannot fit in GPU memory in full precision
+    if config.fp16:
+        main_proc_print(" > use fp16...")
+        model.half()
+    return model
+
+
+def model_to_ddp(model: nn.Module) -> nn.Module:
+    """model_to_ddp"""
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/trainer_adapter.py:34]: 同上

File: training/benchmarks/resnet50/pytorch/train/evaluator.py [hunk 10]
```diff
@@ -1,46 +1,23 @@
...
-            dist.all_reduce(total, dist.ReduceOp.SUM, async_op=False)
-            self.total_loss, self.total_acc1, self.total_acc5, self.total_batch = total.tolist(
-            )
-
-        loss = self.total_loss / self.total_batch
-        acc1 = self.total_acc1 / self.total_batch
-        acc5 = self.total_acc5 / self.total_batch
-        return loss, acc1, acc5
+import torch
+import torch.distributed as dist
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/evaluator.py:2]: dist未使用

File: training/benchmarks/resnet50/pytorch/train/training_state.py [hunk 11]
```diff
@@ -1,84 +1,40 @@
...
+    num_trained_samples = 0
+
+    def status(self):
+        """get status"""
+        if self.converged:
+            self._status = "success"
+        return self._status
+
+    def converged_success(self):
+        """converged success"""
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/train/training_state.py:38]: foo comments

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 12]
```diff
@@ -1,176 +1,134 @@
...
-
+sys.path.append(os.path.abspath(os.path.join(CURR_PATH,
+                                             "../../")))  # benchmarks目录
 # 本地库
 import config
 from driver import Event, dist_pytorch
 from driver.helper import InitHelper
 
-# 导入相关的模块、方法、变量。这里保持名称一致，实现可以不同。
+# TODO 导入相关的模块、方法、变量。这里保持名称一致，实现可以不同。
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:21]: rm TODO

File: training/nvidia/resnet50-pytorch/README.md [hunk 13]
```diff
@@ -15,12 +15,30 @@
...
+| 去除评估部分时间/性能 | t3,见“性能指标”        | 不包含每个epoch末尾的评估部分    |
+| 计算部分时间/性能     | t4,见“性能指标”        | 不包含数据IO部分                 |
+| 训练结果              | acc,见“性能指标”       | 单位为top1分类准确率(acc1)       |
+| 额外修改项            | 无                     |                                  |
+
+* 性能指标
+
+| 配置               | precision | bs   | t1   | t2   | t3   | t4 | acc  | mem |
+| ------------------ | --------- | ---- | ---- | ---- | ---- | ---- |  ---- | ---- |
+| A100单机8卡（1x8） | fp32 | 32 |      |      |      |      |      | 5.8/40.0 |
```
Comment (yuzhou03) [inline training/nvidia/resnet50-pytorch/README.md:42]: t1~t4待更新

File: training/nvidia/resnet50-pytorch/config/config_common.py [hunk 14]
```diff
@@ -1,2 +0,0 @@
-# DDP type: 'apex' or 'native'.
```
Comment (yuzhou03) [inline training/nvidia/resnet50-pytorch/config/config_common.py:1]: 这里不需要设置 vendor = 'nvidia' ?

**Review by yuzhou03 — APPROVED** [review]

**Review by upvenly — APPROVED** [review]

**Pull request merged**